sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -682,6 +684,68 @@ private void writeRequestWithInjectedDelay(
         );
     }
 
+    private void completeWithInjectedDelay(
+        final ChannelHandlerContext context,
+        final RntbdRequestRecord rntbdRequestRecord,
+        final StoreResponse storeResponse,
+        Duration delay) {
+
+        long actualTransitTime = Duration.between(rntbdRequestRecord.timeCreated(), Instant.now()).toNanos();
+        if (actualTransitTime + delay.toNanos() > this.tcpNetworkRequestTimeoutInNanos) {
+
+            context.executor().schedule(
```

Comment:
we do not really need to call `expire()` explicitly, because if we do not complete the record within the timeout config, the` expire()` will automatically be invoked

Reply:
ACK